### DISCUSSION TITLE
How tauri integrates multiple front-end frameworks

### DISCUSSION THREAD

**Discussion by wkl007** [tauri-apps/tauri#5035] [Q&A] [answered]
I have an old project using `electron+vue` and now a new project using `tauri+react`. Now I want to integrate these two projects into one and package them with `tauri`. There are two front-end frameworks involved here. How to deal with it?

**Comment by FabianLars** [comment, marked as answer]
If they are using the same frontend bundler, you _may_ be able to use a feature like this https://vitejs.dev/guide/build.html#multi-page-app. Not 100% sure if it works with different frameworks tbh, but if it works this would make it super easy, especially setting up the devPath.

In a more general sense: You basically just need to make sure that both frontends land in the same `distDir` and then you need to set each window's url accordingly, relative to the distDir.
Setting the `devPath` + window urls for development may be a little tricky tho (unless one of your frontends can proxy to the other). You probably have to play around with using multiple tauri configs (one for dev and one for build) to set the specific window urls to the dev server urls 🤔 (btw you can specify the path to a config you want to use as an overwrite with the `-c`/`--config` cli flag)

**Reply by wkl007** [reply]
Thanks for the reply, I will try it.
